wordpress-mobile/WordPress-Android
# Prepublishing Nudges : Refactor Visibility to match Post Status

Fixes #11889

## Solution
Previously, we only had three `Visibility` states. Public, Private & Password Protected. However, during testing it was realized that there would be issues when trying to map each `Visibility` to the `PostStatus` and a post that is `Public` could mean many things so, for consistency and maintainability, the `PostStatus` behavior in Post Settings was replicated. 

## Testing
1. Create a new post. 
2. Click Publish. 
3. Click the Visibility. 
4. Test each of them to ensure changing them does the same in Post Settings. 

Before | After 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/1509205/82715052-02209900-9c57-11ea-944a-63cad8cc7bcc.png" width="320"></kbd>  |       <kbd><img src="https://user-images.githubusercontent.com/1509205/82715054-0351c600-9c57-11ea-8f59-bc07bd2682dc.png" width="320"></kbd>

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
